moveit_experimental/moveit_jog_arm/src/jog_interface_base.cpp

```diff
@@ -91,10 +91,10 @@ bool JogInterfaceBase::readParameters(ros::NodeHandle& n)
   // parameter was removed, replaced with separate self- and scene-collision proximity thresholds; if old parameter
   // exists, print warning and use old parameter for both new parameter values
   // TODO(JStech): remove this deprecation warning in ROS Noetic
-  if (!rosparam_shortcuts::get("", n, parameter_ns + "/self_collision_proximity_threshold",
-                               ros_parameters_.self_collision_proximity_threshold) &&
-      !rosparam_shortcuts::get("", n, parameter_ns + "/scene_collision_proximity_threshold",
-                               ros_parameters_.scene_collision_proximity_threshold))
+  if (!(rosparam_shortcuts::get("", n, parameter_ns + "/self_collision_proximity_threshold",
+                                ros_parameters_.self_collision_proximity_threshold) &&
+        rosparam_shortcuts::get("", n, parameter_ns + "/scene_collision_proximity_threshold",
+                                ros_parameters_.scene_collision_proximity_threshold)))
   {
     if (rosparam_shortcuts::get("", n, parameter_ns + "/collision_proximity_threshold",
```

Comment:
I think it makes sense to always print the deprecation warning, even if both new parameters are defined. How about you move this parameter lookup before the new parameters and simply override the values if the new parameters are defined?

Reply:
OK. Took me a bit to get all the cases how I like them, but here's where it stands:

- If the old parameter is defined, the deprecation warning is printed, regardless of whether zero, one, or both new parameters are defined;
- If the old parameter is defined and one or both new parameters are missing, the old parameter is used in their place;
- If, after all that, one or both new parameters is not defined, we shutdown via `rosparams_shortcuts::shutdownIfError`;
- Also, no error is printed if the old parameter is missing, but errors are printed if the new parameters are missing.